src/plugins/data/public/query/query_string/dataset_service/types.ts

```diff
@@ -117,5 +117,5 @@ export interface DatasetTypeConfig {
   /**
    * Returns the initial query that is added to the query editor when a dataset is selected.
    */
-  getInitialQueryString?: (query: Query) => string | void;
+  getInitialQueryString?: (query: Query, sampleSize?: number) => string | void;
```

Comment:
woule it be worth either pass

```suggestion
  getInitialQueryString?: (query: Query, services?: Partial<IDataPluginServices>) => string | void;
```